app/models/user_column.rb

```diff
@@ -0,0 +1,40 @@
+class UserColumn < ApplicationRecord
+  include Exportable
+  belongs_to :user
+  validates :user_id, :resource, :presence => true
+  serialize :columns
+  attr_exportable :resource, :columns
+  validates :resource, :inclusion => {:in => ::UserColumnRegistry::Manager.resources.map(&:to_s)}
+  validate :ensure_columns_valid, :if => proc { |r| r.columns.present? && r.resource.present? }
+
+  def ensure_columns_valid
+    valid_columns = ::UserColumnRegistry::Manager.columns(resource) || []
+    valid_column_names = valid_columns.map(&:name)
+    invalid_columns = columns.select{ |col| !valid_column_names.include? col.to_s }
+    if invalid_columns.present?
+      errors.add(:columns, N_("invalid column names '%s' specified for the given resource.") % invalid_columns.inspect)
+    end
+  end
+
+  def self.formatted_columns(resource)
+    registry_columns = UserColumnRegistry::Manager.columns(resource)
+    return [] if registry_columns.blank?
+
+    current_user_columns = nil
+
+    if User.current.present?
+      uc = User.current.user_columns.where(:resource => resource).first
```

Comment:
nitpick: I find variable ```uc``` name confusing, how about simple columns?
consider using scope instead of where.